sys/xtimer/xtimer_core.c

```diff
@@ -281,32 +240,13 @@ static inline void _update_short_timers(uint64_t *now)
  */
 static void _timer_callback(void)
 {
-    uint64_t now;
+    uint64_t now = _xtimer_now64();
     _in_handler = 1;
     _lltimer_ongoing = false;
-    now = _xtimer_now64();
-
-update:
-    /* update short timer offset and fire */
-    _update_short_timers(&now);
-    /* update long timer offset */
-    _update_long_timers(&now);
-    /* update current time */
-    now = _xtimer_now64();
-
-    if (timer_list_head) {
-        /* make sure we're not setting a time in the past */
-        uint32_t elapsed = (uint32_t)now - timer_list_head->start_time;
-        if (timer_list_head->offset < elapsed ||
-            timer_list_head->offset - elapsed < XTIMER_ISR_BACKOFF) {
-            goto update;
-        }
-        else {
-            timer_list_head->offset -= elapsed;
-            timer_list_head->start_time = (uint32_t)now;
-            timer_list_head->long_start_time = (uint32_t)(now >> 32);
-        }
-    }
```

Comment:
_update_timers already did timer expiry check and it does not have to be re-done here.